lib/components/search.jsx

```diff
@@ -0,0 +1,202 @@
+'use strict'
+
+const figures = require('figures')
+const {
+  h,
+  Color,
+  Component,
+  Fragment,
+  Text
+} = require('ink')
+const SelectInput = require('ink-select-input')
+const Spinner = require('ink-spinner')
+const TextInput = require('ink-text-input')
+const libnpm = require('libnpm')
+
+
+const MAX_RESULTS = 10
+
+
+class Search extends Component {
+  constructor (props) {
+    super(props)
+    const { terms } = this.props
+    this.state = {
+      isInstalling: false,
+      isLoading: null,
+      matches: [],
+      selectedPackage: null,
+      terms: terms || ''
+    }
+    this.onChangeTerms = this.onChangeTerms.bind(this)
+    this.onSubmit = this.onSubmit.bind(this)
+    this.onSelectPackage = this.onSelectPackage.bind(this)
+  }
+
+  componentDidMount () {
+    const { terms } = this.props
+    if (terms) {
+      this.search(terms)
+    }
+  }
+
+  render () {
+    const {
+      isInstalling,
+      isLoading,
+      matches,
+      selectedPackage,
+      terms
+    } = this.state
+
+    return <div>
+      <SearchInput
+        terms={terms}
+        onChange={this.onChangeTerms}
+        onSubmit={this.onSubmit} />
+      <SearchResults
+        isLoading={isLoading}
+        terms={terms}
+        matches={matches}
+        onSelect={this.onSelectPackage} />
+      { isInstalling && selectedPackage
+        ? <InstallingPackage isInstalling={isInstalling} pkg={selectedPackage} />
+        : null
+      }
+    </div>
+  }
+
+  async search (terms) {
+    try {
+      this.setState({
+        isLoading: true,
+        matches: []
+      })
+      const matches = await libnpm.search(terms, {
+        detailed: true,
+        limit: MAX_RESULTS
+      })
+      this.setState({
+        isLoading: false,
+        matches
+      })
+    } catch (err) {
+      // @todo Show error message
+      this.setState({
+        isLoading: false,
+        matches: []
+      })
+    }
+  }
+
+  async install (pkg) {
+    try {
+      this.setState({
+        isInstalling: true,
+        selectedPackage: pkg
+      })
+      // @todo Implement real install
+      await new Promise((resolve, reject) => setTimeout(resolve, 1000))
+      this.setState({ isInstalling: false })
+      this.props.onExit()
+    } catch (err) {
+      // @todo Show error message
+      this.setState({ isInstalling: false })
+    }
+  }
+
+  onChangeTerms (terms) {
+    this.setState({ terms })
+  }
+
+  onSubmit (terms) {
+    this.search(terms)
+  }
+
+  onSelectPackage (pkg) {
+    this.install(pkg)
+  }
+}
+
+const SearchInput = ({ onChange, onSubmit, terms }) => {
+  return <div>
+    <Color grey bold>Find a package: </Color>
+    <TextInput value={terms} onChange={onChange} onSubmit={onSubmit} />
+  </div>
```

Comment:
For some reason, if I erase the textinput data and press "Enter" it does BOTH a search and an install (and closes the process). Or was I testing an older version?

Reply:
Yea! This is that weird input focus thing I mentioned in my Todos list. I saw that the `ink-select-input` has an explicit `focus` boolean prop, and I just realized `ink-text-input` has the same prop undocumented. I just need to add some logic to juggle these intuitively...